src/modules/fstab/main.py

```diff
@@ -252,17 +253,28 @@ def generate_fstab_line_info(self, partition):
             self.root_is_ssd = is_ssd
 
         if filesystem == "btrfs" and "subvol" in partition:
-            return dict(
-                device="UUID=" + partition["uuid"],
-                mount_point=mount_point,
-                fs=filesystem,
-                options=",".join(
-                    ["subvol={}".format(partition["subvol"]), options]
-                    ),
-                check=check,
-                )
-
-        if "luksMapperName" in partition:
+            if has_luks:
```

Comment:
There must be a better way. By now we have four giant `return dict()` statements, which differ only in details.